Orange/data/table.py

```diff
@@ -112,26 +112,33 @@ def __setitem__(self, key, value):
                 self.table.metas[self.row_index, -1 - key] = value
 
     def _str(self, limit):
-        def sp_values(matrix, variables):
+        def sp_values(matrix, variables, show_labels=True):
             if not sp.issparse(matrix):
                 return Instance.str_values(matrix[row], variables, limit)
-            begptr, endptr = matrix.indptr[row:row + 2]
-            rendptr = endptr if not limit else min(endptr, begptr + 5)
-            variables = [variables[var]
-                         for var in matrix.indices[begptr:rendptr]]
+
+            # Sparse matrices can't handle slices where an index is out of
+            # bounds like numpy arrays can, so we must determine largest index
+            _, columns = matrix.shape
+            max_idx, has_more_columns = min(5, columns), columns > 5
+
             s = ", ".join(
-                "{}={}".format(var.name, var.str_val(val))
-                for var, val in zip(variables, matrix.data[begptr:rendptr]))
-            if limit and rendptr != endptr:
+                ("%s=" % var.name if (var.is_discrete or matrix[row, idx]) and
```

Comment:
We probably misunderstood. I wasn't objecting to the `interface=1.000` part. The problem is in the following `0.000` part for which I do not know which feature it represents. Also, we should probably skip showing zeros in sparse data altogether (which we do not in the above example).

I suggest we stick to the format that is used in Table. That is, for all values that are non-zero we show `<feature>=<value>`. We skip printing features with zero values altogether, except for discrete features when we print `<feature>=<value>` pair regardless of the value.